File: PyProject/leetcode/rotate-array.py
from typing import List


class RotateArray:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        self.reversed(nums, 0, len(nums) - 1)
        self.reversed(nums, 0, k - 1)
        self.reversed(nums, k, len(nums) - 1)

    def reversed(self, nums: List[int], start, end):
        while start < end:
            # 交换
            tmp = nums[start]
            nums[start] = nums[end]
            nums[end] = tmp
            start += 1
            end -= 1

File: GoProject/leetcode/array/rotate-array.go
package array

func rotate(nums []int, k int) {

	// 移动一个

	l := len(nums)

	for j:=0;j<k;j++ {
		tmp := nums[l-1]

		for i := 0; i < l-1; i++ {
			nums[l-1-i] = nums[l-2-i]
		}
		nums[0] = tmp
	}

}
